### DISCUSSION TITLE
Get user avatar from Azure provider (or oauth provider) automatically?

### DISCUSSION THREAD

**Discussion by gouku** [outline/outline#3463] [Self Hosting]
After setting up Outline with Azure AD I found that user avatars are not automatically pulled from M365. Is there any way to do this?

**Comment by tommoor** [comment]
The code is included in the Azure integration to bring across profile pictures, if it didn't it might be representative of your storage (S3/Minio) not being setup correctly.

**Reply by gouku** [reply]
Thanks for reply! Yes I did setup Azure AD first then set MinIO after provisioning users. Is there anyway to sync user info (avatar, etc.) after provision?

**Reply by tommoor** [reply]
Not currently, although it would be nice to have this option

**Comment by samstride** [comment]
> Not currently, although it would be nice to have this option

@tommoor , any updates on being able to sync profile? We are using gitlab as an auth provider and would benefit from this feature.

Thanks.

**Reply by tommoor** [reply]
Nothing has changed here, the reason it wasn't done originally is because we don't want to overwrite any changes that the user might have made to their profile. I suppose we could track if it's been manually updated or have an explicit option to keep it synced, neither of these feel _great_ though

**Comment by samstride** [comment]
@tommoor , thanks. What about syncing only the avatar and not the full profile?

**Reply by tommoor** [reply]
The same issue applies, we need to make sure that:

a) we're not overriding any avatar the user may have uploaded on their profile
b) we're not uploading a new image on every sign-in – so we need a way to tell if it's actually changed

These are both solvable of course

**Reply by samstride** [reply]
Thanks. What about providing a button that the user needs to click to sync the avatar only. That way it's user controlled/driven (similar to the upload avatar) ?
